arm-machinelearning/2016-05-01-preview/swagger/commitmentPlans.json

```diff
@@ -0,0 +1,720 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure ML Commitment Plans Management Client",
+    "description": "These APIs allow end users to operate on Azure Machine Learning Commitment Plans resources and their child Commitment Association resources. They support the following operations:<ul><li>Create or update a commitment plan</li><li>Get a commitment plan</li><li>Patch a commitment plan</li><li>Delete a commitment plan</li><li>Get all commitment plans in a resource group</li><li>Get all commitment plans in a subscription</li><li>Get a commitment association</li><li>Get all commitment associations for a commitment plan</li><li>Move a commitment association from one commitment plan to another</li><li>Get usage history for a commitment plan</li></ul>",
+    "version": "2016-05-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearning/commitmentPlans/{planName}/commitmentAssociations/{associationName}": {
+      "get": {
+        "tags": [
+          "CommitmentAssociations"
+        ],
+        "operationId": "CommitmentAssociations_Get",
+        "description": "Gets a commitment association.",
+        "parameters": [
+          {
+           "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
```

Comment:
Rather than repeating the resourceGroupName definition everywhere, it can be defined once in the global parameters section with the extension `"x-ms-parameter-location": "method"` and referenced every where like SubscriptionIdParameter and ApiVersionParameter. That would be easier to maintain in the long run